tests/bluetooth/tester/src/btp_bap.c

```diff
@@ -17,35 +18,1258 @@
+	btp_send_ascs_operation_completed_ev(stream->conn, stream->ep->status.id,
+					     BT_ASCS_START_OP, 0);
+}
+
+static void stream_stopped(struct bt_bap_stream *stream, uint8_t reason)
+{
+	LOG_DBG("Stopped stream %p with reason 0x%02X", stream, reason);
+	btp_send_ascs_operation_completed_ev(stream->conn, stream->ep->status.id,
+					     BT_ASCS_STOP_OP, 0);
+}
+
+static void stream_recv(struct bt_bap_stream *stream,
+			const struct bt_iso_recv_info *info,
+			struct net_buf *buf)
+{
+	LOG_DBG("Incoming audio on stream %p len %u", stream, buf->len);
+}
+
+static void stream_sent(struct bt_bap_stream *stream)
+{
+	LOG_DBG("Stream %p sent", stream);
+}
+
+static struct bt_bap_stream_ops stream_ops = {
+	.configured = stream_configured,
+	.qos_set = stream_qos_set,
+	.enabled = stream_enabled,
+	.metadata_updated = stream_metadata_updated,
+	.disabled = stream_disabled,
+	.released = stream_released,
+	.started = stream_started,
+	.stopped = stream_stopped,
+	.recv = stream_recv,
+	.sent = stream_sent,
+};
+
+static void btp_send_pac_codec_found_ev(struct bt_conn *conn, const struct bt_codec *codec,
+					enum bt_audio_dir dir, uint16_t attr_handle)
+{
+	struct btp_bap_codec_cap_found_ev ev;
+	struct bt_conn_info info;
+	const struct bt_codec_data *data;
+
+	bt_conn_get_info(conn, &info);
+	bt_addr_le_copy(&ev.address, info.le.dst);
+
+	ev.dir = dir;
+	ev.coding_format = codec->id;
+	ev.attr_handle = attr_handle;
+
+	bt_codec_get_val(codec, BT_CODEC_LC3_FREQ, &data);
+	memcpy(&ev.frequencies, data->data.data, sizeof(ev.frequencies));
+
+	bt_codec_get_val(codec, BT_CODEC_LC3_DURATION, &data);
+	memcpy(&ev.frame_durations, data->data.data, sizeof(ev.frame_durations));
+
+	bt_codec_get_val(codec, BT_CODEC_LC3_FRAME_LEN, &data);
+	memcpy(&ev.octets_per_frame, data->data.data, sizeof(ev.octets_per_frame));
+
+	tester_event(BTP_SERVICE_ID_BAP, BTP_BAP_EV_CODEC_CAP_FOUND, &ev, sizeof(ev));
+}
+
+static void btp_send_ase_found_ev(struct bt_conn *conn, struct bt_bap_ep *ep, uint16_t attr_handle)
+{
+	struct btp_ascs_ase_found_ev ev;
+	struct bt_conn_info info;
+
+	bt_conn_get_info(conn, &info);
+	bt_addr_le_copy(&ev.address, info.le.dst);
+
+	ev.ase_id = ep->status.id;
+	ev.dir = ep->dir;
+	ev.attr_handle = attr_handle;
+
+	tester_event(BTP_SERVICE_ID_BAP, BTP_BAP_EV_ASE_FOUND, &ev, sizeof(ev));
+}
+
+static void discover_remote_ases_cb(struct bt_conn *conn, struct bt_codec *codec,
+				    struct bt_bap_ep *ep,
+				    struct bt_bap_unicast_client_discover_params *params)
+{
+	int err;
+	enum bt_audio_dir dir = params->dir;
+	struct audio_connection *audio_conn;
+
+	if (params->err != 0 && params->err != BT_ATT_ERR_ATTRIBUTE_NOT_FOUND) {
+		LOG_DBG("Discover remote ASEs failed: %d", params->err);
+		return;
+	}
+
+	if (codec != NULL) {
+		LOG_DBG("Discovered codec capabilities %p", codec);
+		print_codec(codec);
+		btp_send_pac_codec_found_ev(conn, codec, dir, params->read.single.handle);
+
+		return;
+	}
+
+	if (ep != NULL) {
```

Comment:
I'm refactoring the BAP discovery procedure in https://github.com/zephyrproject-rtos/zephyr/pull/56108. 

Is this code built in CI, so that in case this gets in before my PR, then my PR would fail CI?

Reply:
After some refactors I do not need to pass the handlers as much, so removed this parameter. It saved me just a little scan time.